src/main/resources/static/data-mapping.js

```diff
@@ -16,122 +23,143 @@ $( document ).on( "mousemove", function( event ) {
+        d3.json('data/test-oc-data.json', function (data) {
+            oc_data = data;
+            visualizeOCTree(oc_data);
+        });
+        d3.json('data/test-map-data.json', function(data) {
+            map_data = data;
+        });
     });
 
+    //for testing: mapping data
     $('#auto-map-btn').click(function(){
-        d3.tsv('test-mapping.tsv', function(mapping) {
-            uploaded_mapping = mapping;
-            for(var i=0; i<mapping.length; i++) {
-                var inst = mapping[i];
-                var usr_name = inst['USR'];
-                var oc_name = inst['OC'];
-                clearManualItems(usr_name);
-                for(var j=0; j<ocitems.length; j++) {
-                    var item = ocitems[j];
-                    if(item.ocname == oc_name) {
-                        item.connected = true;
-                        item.usrname = usr_name;
-                        break;
-                    }
-                }//for each mapping-pair in oc-items
-            }//for each mapping instance
-            updateMatchingItems();
-            var is_tree_view = $('#tree-toggle').prop('checked');
-            if(!is_tree_view) {
-                d3.selectAll('.mitem').each(function(d,i){
-                    if(d.connected) {
-                        var m = 5;
-                        d3.selectAll('.griditem').each(function(gridd, gridi) {
-                            if(gridd.name == d.usrname) {
-                                gridd.x = +d.x - 2*m - d.w;
-                                gridd.y = +d.y;
-                                gridd.mapped = true;
-                                gridd.matchingItem = d;
+        clearMapping();
+        for(var i=0; i<map_data.length; i++) {
+            var pair = map_data[i];
+            var ocd = findOCitem(pair['Study_Site'], pair['EventName'], pair['CRFname'], pair['CRFversion'], pair['ocItemName']);
+            var usritem_obj = findUsrItem(pair['usrItemName']);
+            // console.log(ocd);
+
+            if(ocd !== null && usritem_obj !== null) {
+                var ocname = ocd.name;
+                var ocCRFv = ocd.parent.name;
+                var ocCRF = ocd.parent.parent.name;
+                var ocEventName = ocd.parent.parent.parent.name;
+                var ocStudy = ocd.parent.parent.parent.parent.name;
+                var path = ocStudy+"\t"+ocEventName+"\t"+ocCRF+"\t"+ocCRFv+"\t"+ocname;
+                var d = usritem_obj.datum();
+                if(mapped_ocitems.indexOf(path) == -1) {
+                    mapped_ocitems.push(path);
+                    d.mapped = true;
+                    d.ocItemName = ocname;
+                    d.ocCRFv = ocCRFv;
+                    d.ocCRF = ocCRF;
+                    d.ocEventName = ocEventName;
+                    d.ocStudy = ocStudy;
+                    d.ocItemData = ocd;
+                    d.ocPath = path;
+                }
+                positionUsrList(usr_item_data, 800);
```

Comment:
Can you make it into named variable then? In case you use it in multiple places it will be more convenient to change one variable than to look-up 800 for instance. Plus it will be more readable.